### ISSUE TITLE
Systematic false positive: Roller coaster tracks

### ISSUE THREAD

**Issue by Hiausirg** [osm-fr/osmose-backend#1995]
Hello,
the check "Untagged named object" ([2110](https://osmose.openstreetmap.fr/en/issues/open?item=2110)) highlights objects which are (correctly) tagged with "roller_coaster=track" as being "untagged". This should be fixed.
I assume that roller coasters which have no name are probably also catched as untagged by another filter/check, but I don't know which one.
I noticed this issue [here](https://osmose.openstreetmap.fr/en/map/#zoom=16&lat=34.52024&lon=-93.01494&item=xxxx&level=1%2C2).

Note that those roller coasters are also highlighted as being self-intersecting polygons. I'm unsure if this is an issue (but I think that it is very common that roller coasters overlap)

Greetings

**Comment by Famlam** [comment]
> the check "Untagged named object" ([2110](https://osmose.openstreetmap.fr/en/issues/open?item=2110)) highlights objects which are (correctly) tagged with "roller_coaster=track" as being "untagged". This should be fixed.

I'm not sure.. is the track itself named (like a highway) or should the name tag be on the surrounding area ( https://wiki.openstreetmap.org/wiki/Tag:attraction%3Droller_coaster ) that also includes i.e. the buildings etc that belong to it. It seems to be used quite... randomly if I just peak around some attraction parks.

> Note that those roller coasters are also highlighted as being self-intersecting polygons. I'm unsure if this is an issue (but I think that it is very common that roller coasters overlap)

Oops, when fixing https://github.com/osm-fr/osmose-backend/pull/1960 I forgot to update https://github.com/osm-fr/osmose-backend/blob/08b919d02c0c0e8c7729261e88886fd92d9b554c/analysers/analyser_osmosis_polygon.py#L35
However, I suspect that roller coasters should have different layers where they overlap. So I guess it's better to split the way in ways with different layers anyway.

**Comment by Famlam** [comment]
@frodrigo your call, allow name on rollercoaster tracks or not? I have no strong opinion, and it is pretty common

**Comment by frodrigo** [comment]
As say on the wiki, for me the name should be on with `attraction=roller_coaster` key.

`attraction=roller_coaster` + `roller_coaster=track` Could be on same object, with the name ? That may solve the issue

**Comment by Famlam** [comment]
> attraction=roller_coaster + roller_coaster=track Could be on same object [...]

No, the related wikis explicitly mention that those two tags are incompatible. (Even though it occurs relatively frequently)

**Comment by Famlam** [comment]
@Hiausirg do you have a strong opinion about this? Maybe it was discussed in a mailing list that we're not aware of, for instance

**Comment by Hiausirg** [comment]
Not at all. Track, attraction or both?, no idea.
I've never mapped any such theme parks or similar, the primary problem I saw was that those roller coaster tracks were highlighted as being "untagged" objects, but "roller_coaster=track" is a 'primary' descriptive tag (so they're not actually "untagged"). Hope thats understandable.

No idea where the name should be.

**Comment by Famlam** [comment]
How about simply `Unexpected name on element` instead of `Untagged named object`?

**Comment by frodrigo** [comment]
> How about simply `Unexpected name on element` instead of `Untagged named object`?

I think, the issue is more about missing tags than the name tag.

**Comment by Famlam** [comment]
True, I was primarily thinking of something that would also cover existing objects (like the roller coaster track) that are tagged, but not expected to have a name on it. `untagged named object` is in that sense incorrect for such cases.
